### PR TITLE
Support NumPy 1.13

### LINKED ISSUE #137: Support numpy 1.13

### PR DESCRIPTION
**Pull request by okuta**
Fix #137 
- Use `numpy.AxisError`.
- Skip bool test case.

### CODE REVIEW

File: cupy/manipulation/rearrange.py [hunk 1]
```diff
@@ -23,7 +25,8 @@ def flip(a, axis):
 
     axis = int(axis)
     if not -a_ndim <= axis < a_ndim:
-        raise ValueError('axis must be >= %d and < %d' % (-a_ndim, a_ndim))
+        raise ValueError(
+            'axis must be >= %d and < %d' % (-a_ndim, a_ndim))
```
Comment (gwtnb) [inline cupy/manipulation/rearrange.py]: Is it intended to use `ValueError` instead of `numpy.AxisError`?

Comment (okuta) [inline cupy/manipulation/rearrange.py]: Yes, this is same as NumPy.

Comment (gwtnb) [inline cupy/manipulation/rearrange.py]: ok

File: cupy/statistics/histogram.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def bincount(x, weights=None, minlength=None):
         raise ValueError('The weights and list don\'t have the same length.')
     if minlength is not None:
         minlength = int(minlength)
-        if minlength <= 0:
+        if minlength < 0:
```
Comment (unnonouno) [inline cupy/statistics/histogram.py]: Is this bug or change in the latest numpy?

Comment (okuta) [inline cupy/statistics/histogram.py]: Yes. NumPy v1.12.1 raises error.
```
======================================================================
FAIL: test_bincount_zero_minlength (test_histogram.TestHistogram)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/okuta/env/chainer/cupy/cupy/testing/helper.py", line 484, in test_func
    impl(self, *args, **kw)
  File "/home/okuta/env/chainer/cupy/cupy/testing/helper.py", line 461, in test_func
    accept_error=accept_error)
  File "/home/okuta/env/chainer/cupy/cupy/testing/helper.py", line 45, in _check_cupy_numpy_error
    self.fail('Only numpy raises error\n\n' + numpy_tb)
AssertionError: Only numpy raises error
```

File: tests/cupy_tests/statics_tests/test_histogram.py [hunk 3]
```diff
@@ -94,4 +94,4 @@ def test_bincount_too_small(self, xp, dtype):
     @testing.numpy_cupy_raises()
     def test_bincount_zero_minlength(self, xp, dtype):
```
Comment (unnonouno) [inline tests/cupy_tests/statics_tests/test_histogram.py]: Change this test name, and add test case for minlength=0.

File: cupy/statistics/histogram.py (hunk 2 above)
Comment (unnonouno) [inline cupy/statistics/histogram.py]: OK, this change. https://github.com/numpy/numpy/commit/ec6d4295a80e5df235d3f5445e6425581309c930

File: tests/cupy_tests/statics_tests/test_histogram.py (hunk 3 above)
Comment (gwtnb) [inline tests/cupy_tests/statics_tests/test_histogram.py]: @okuta Could you fix before the next release?

**Comment by okuta** [comment]
I fixed tests

**Comment by unnonouno** [comment]
LGTM

**Comment by unnonouno** [comment]
Its behavior is changed. We cannot backport this PR.

**Pull request merged**